Adapter/PhpcrOdmAdapter.php

```diff
@@ -190,12 +212,49 @@ public function getReferringAutoRoutes($contentDocument)
     public function findRouteForUri($uri, UriContext $uriContext)
     {
         $path = $this->getPathFromUri($uri);
+        $document = $this->dm->find(null, $path);
 
-        return $this->dm->find(null, $path);
+        if ($document instanceof AutoRouteInterface) {
+            return $document;
+        }
+
+        return null;
     }
 
     private function getPathFromUri($uri)
     {
         return $this->baseRoutePath . $uri;
     }
+
+    /**
+     * Convert the given generic document to an auto route document, migrating any children which it may have.
+     *
+     * @param Generic $document
+     * @param object $contentDocument
+     * @param string $autoRouteTag
+     * @param string $routeType
+     * @return AutoRouteInterface
+     */
+    private function migrateGenericToAutoRoute(Generic $document, $contentDocument, $autoRouteTag, $routeType)
+    {
+        $autoRouteClassName = $this->autoRouteFqcn;
```

Comment:
+1 for keeping this for now

On Wed, Nov 04, 2015 at 11:54:10PM -0800, Manuele Menozzi wrote:
>    In [1]Adapter/PhpcrOdmAdapter.php:
> 
>  >      }
>  >
>  >      private function getPathFromUri($uri)
>  >      {
>  >          return $this->baseRoutePath . $uri;
>  >      }
>  > +
>  > +    /**
>  > +     * Convert the given generic document to an auto route document, migrating any children which it may have.
>  > +     *
>  > +     * @param Generic $document
>  > +     * @param object $contentDocument
>  > +     * @param string $autoRouteTag
>  > +     * @param string $routeType
>  > +     * @return AutoRouteInterface
>  > +     */
>  > +    private function migrateGenericToAutoRoute(Generic $document, $contentDocument, $autoRouteTag, $routeType)
>  > +    {
>  > +        $autoRouteClassName = $this->autoRouteFqcn;
> 
>    [2]@dantleech [3]@lsmith77 [4]@dbu I agree. An higher level method which
>    migrates document class should be in Doctrine PHPCR-ODM but I also think
>    that we should merge this PR now in the RoutingAutoBundle and then
>    refactor when this higher level method will be available in Doctrine
>    PHPCR-ODM.
> 
>    —
>    Reply to this email directly or [5]view it on GitHub.
> 
>    Reverse link: [6]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony-cmf/RoutingAutoBundle/pull/165#discussion_r43984768
>    2. https://github.com/dantleech
>    3. https://github.com/lsmith77
>    4. https://github.com/dbu
>    5. https://github.com/symfony-cmf/RoutingAutoBundle/pull/165/files#r43984768
>    6. https://github.com/symfony-cmf/RoutingAutoBundle/pull/165/files#r43984768